rust/otel-arrow-rust/src/otap/filter/logs.rs

```diff
@@ -0,0 +1,788 @@
+            arrow::compute::not(&log_attr_filter).expect("not doesn't fail");
+        let log_attr_parent_ids_removed = self.get_ids(
+            log_attr_parent_ids_column,
+            &inverse_log_attr_filter,
+            consts::PARENT_ID,
+        )?;
+        let log_record_ids_filter =
+            self.build_id_filter(log_record_ids_column, log_attr_parent_ids_removed, true);
+        log_record_filter = arrow::compute::and_kleene(&log_record_filter, &log_record_ids_filter)
+            .expect("can combine two boolean arrays with equal length");
+
+        // now using the updated log_record_filter we need to update the rest of the filers
+        let inverse_log_record_filter =
+            arrow::compute::not(&log_record_filter).expect("not doesn't fail");
+        let log_record_ids_removed = self.get_ids(
+            log_record_ids_column,
+            &inverse_log_record_filter,
+            consts::ID,
+        )?;
+        let log_attr_parent_ids_filter =
+            self.build_id_filter(log_attr_parent_ids_column, log_record_ids_removed, true);
+        log_attr_filter = arrow::compute::and_kleene(&log_attr_filter, &log_attr_parent_ids_filter)
+            .expect("can combine two boolean arrays with equal length");
+
+        // part 4: clean up resource attrs
+
+        let log_record_resource_ids_kept = self.get_ids(
+            log_record_resource_ids_column,
+            &log_record_filter,
+            consts::ID,
+        )?;
+        let resource_attr_parent_ids_filter = self.build_id_filter(
+            resource_attr_parent_ids_column,
+            log_record_resource_ids_kept,
+            false,
+        );
+        resource_attr_filter =
+            arrow::compute::and_kleene(&resource_attr_filter, &resource_attr_parent_ids_filter)
+                .expect("can combine two boolean arrays with equal length");
+
+        let scope_attr_filter = if let Some(scope_attrs_record_batch) = scope_attrs {
+            let scope_attr_parent_ids_column =
+                get_required_array(scope_attrs_record_batch, consts::PARENT_ID)?;
+            let log_record_scope_ids_kept =
+                self.get_ids(log_record_scope_ids_column, &log_record_filter, consts::ID)?;
+            Some(self.build_id_filter(
+                scope_attr_parent_ids_column,
+                log_record_scope_ids_kept,
+                false,
+            ))
+        } else {
+            None
+        };
+
+        Ok((
+            resource_attr_filter,
+            scope_attr_filter,
+            log_record_filter,
+            log_attr_filter,
+        ))
+    }
+
+    fn get_ids(
+        &self,
+        id_column: &Arc<dyn Array>,
+        filter: &BooleanArray,
+        column_type: &str,
+    ) -> Result<HashSet<u16>> {
+        // get ids being removed
+        // error out herre
+        let filtered_ids = arrow::compute::filter(id_column, filter)
+            .expect("can apply predicate on column with equal length");
```

Comment:
If this `expect` can be triggered by a misconfiguration or by malformed data, then it should be turned into an error.
However, if checks have already been performed upstream and logically this `expect` should never be triggered, then it’s fine to keep it.